demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -764,6 +764,75 @@ CK_RV xProvisionDevice( CK_SESSION_HANDLE xSession,
     return xResult;
 }
 
+CK_RV xInitializePkcs11Token()
+{
+    CK_RV xResult;
+
+    CK_FUNCTION_LIST_PTR pxFunctionList;
+    CK_SLOT_ID * pxSlotId = NULL;
+    CK_ULONG xSlotCount;
+    CK_FLAGS xTokenFlags = 0;
+    CK_TOKEN_INFO_PTR pxTokenInfo = NULL;
+
+    xResult = C_GetFunctionList( &pxFunctionList );
+
+    if( xResult == CKR_OK )
+    {
+        xResult = xInitializePKCS11();
+    }
+
+    if( ( xResult == CKR_OK ) || ( xResult == CKR_CRYPTOKI_ALREADY_INITIALIZED ) )
+    {
+        xResult = xGetSlotList( &pxSlotId, &xSlotCount );
+    }
+
+    if( xResult == CKR_OK )
+    {
+        /* Check if token is initialized */
+        pxTokenInfo = pvPortMalloc( sizeof( CK_TOKEN_INFO ) );
+
+        if( pxTokenInfo != NULL )
+        {
+            /* We will take the first slot available.
+             * If your application has multiple slots, insert logic
+             * for selecting an appropriate slot here.
+             * TODO: Consider a define here instead.
+             */
+            xResult = pxFunctionList->C_GetTokenInfo( pxSlotId[ 0 ], pxTokenInfo );
+        }
+        else
+        {
+            xResult = CKR_HOST_MEMORY;
+        }
+    }
+
+    if( CKR_OK == xResult )
+    {
+        xTokenFlags = pxTokenInfo->flags;
+    }
+
+    if( ( CKR_OK == xResult ) && !( CKF_TOKEN_INITIALIZED & xTokenFlags ) )
+    {
+        /* If token is not initialized - initialize it */
```

Comment:
"If the token is not initialized, initialize it."

Reply:
fixed.